src/utils/cache.ts

```diff
@@ -106,13 +107,9 @@ export const createCacheHelper = <Data = any, ExtendedInfo = {}>(
 ) =>
   [
     // Getter
-    () => cache.get(key) || {},
+    () => (cache.get(key) || {}) as State<Data, any> & Partial<ExtendedInfo>,
```

Comment:
This was necessary to make `ExtendedInfo` work:

https://github.com/vercel/swr/blob/09d80c525798e85e6fd7e1dbc939b68249ad2b29/infinite/index.ts#L67-L77

I couldn't figure out a way to do it without type assertion.

I did notice that this line had a type error (`Type 'Data' is not assignable to type 'Data[]'`), but I couldn't figure out how to resolve this.

https://github.com/vercel/swr/blob/09d80c525798e85e6fd7e1dbc939b68249ad2b29/infinite/index.ts#L147